src/dask_histogram/core.py

```diff
@@ -729,41 +732,34 @@ def _reduction(
     ph: PartitionedHistogram,
     split_every: int | None = None,
 ) -> AggHistogram:
-    from dask.bag.core import empty_safe_aggregate
+    from dask.layers import DataFrameTreeReduction
 
     if split_every is None:
         split_every = dask.config.get("histogram.aggregation.split_every", 8)
     if split_every is False:
         split_every = ph.npartitions
 
     token = tokenize(ph, sum, split_every)
-    name = f"hist-aggregate-{token}"
-    k = ph.npartitions
-    b = ph.name
-    d = 0
-    dsk = {}
-    while k > split_every:
-        c = f"{name}{d}"
-        for i, inds in enumerate(partition_all(split_every, range(k))):
-            dsk[(c, i)] = (
-                empty_safe_aggregate,
-                sum,
-                [(b, j) for j in inds],
-                False,
-            )
-        k = i + 1
-        b = c
-        d += 1
-    dsk[(name, 0)] = (
-        empty_safe_aggregate,
-        sum,
-        [(b, j) for j in range(k)],
-        True,
+
+    label = "histreduce"
+
+    name_comb = f"{label}-combine-{token}"
+    name_agg = f"{label}-agg-{token}"
+
+    dftr = DataFrameTreeReduction(
+        name=name_agg,
+        name_input=ph.name,
+        npartitions_input=ph.npartitions,
+        concat_func=sum,
```

Comment:
Here if we instead use:
```python3
    def hist_safe_sum(items):
        safe_items = [item for item in items if not isinstance(item, tuple)]
        return sum(safe_items)
    
    dftr = DataFrameTreeReduction(
        name=name_agg,
        name_input=ph.name,
        npartitions_input=ph.npartitions,
        concat_func=hist_safe_sum,
        ....    
```

Then we can get away with doing the column optimization on the first partition of a dataset alone. This seems reasonably safe to me?

Reply:
yea seems fine

Reply:
I guess the better way to do it would be to test for what it should be.

I guess that's `bh.Histogram` and then whatever the type for numpy is (which I do not know off the top of my head).